apps/condo/domains/acquiring/schema/CreatePaymentByLinkService.js

```diff
@@ -0,0 +1,250 @@
+                    account: { number: PersAcc, deletedAt: null },
+                    receiver: { bankAccount: PersonalAcc, deletedAt: null },
+                    deletedAt: null,
+                }, {
+                    sortBy: ['period_DESC'],
+                    first: 1,
+                })
+
+                const period = `${PaymPeriod.split('.')[1]}-${PaymPeriod.split('.')[0]}-01`
+
+                let multiPaymentId
+                if (lastBillingReceipt) {
+                    // if period matches we use found receipt to create MultiPayment
+                    if (lastBillingReceipt.period === period) {
+                        const { multiPaymentId: id } = await registerMultiPaymentForOneReceipt(context, {
+                            dv, sender,
+                            receipt: { id: lastBillingReceipt.id },
+                            acquiringIntegrationContext: { id: contextsMap[get(lastBillingReceipt, ['context', 'id'])] },
+                        })
+                        multiPaymentId = id
+                    } else {
+                        if (lastBillingReceipt.period > period) {
+                            // we have a newer receipt
+                            const { multiPaymentId: id } = await registerMultiPaymentForOneReceipt(context, {
+                                dv, sender,
+                                receipt: { id: lastBillingReceipt.id },
+                                acquiringIntegrationContext: { id: contextsMap[get(lastBillingReceipt, ['context', 'id'])] },
+                            })
+                            multiPaymentId = id
+                        } else {
+                            // the last receipt is older than the scanned one
+
+                            // find acquiring context and routing number from older receipt
+                            const billingIntegrationContext = await getById('BillingIntegrationOrganizationContext', lastBillingReceipt.context.id)
+                            /** @type {AcquiringIntegrationContext[]} */
+                            const acquiringContexts = await AcquiringIntegrationContext.getAll(context, {
+                                organization: { id: billingIntegrationContext.organization, deletedAt: null },
+                                status: CONTEXT_FINISHED_STATUS,
+                                deletedAt: null,
+                            })
+                            const bankAccounts = await BankAccount.getAll(context, {
+                                organization: { id: billingIntegrationContext.organization, deletedAt: null },
+                                deletedAt: null,
+                            })
+
+                            const { multiPaymentId: id } = await registerMultiPaymentForVirtualReceipt(context, {
+                                dv, sender,
+                                receipt: {
+                                    currencyCode: bankAccounts[0].currencyCode,
```

Comment:
You need to use currencyCode from acquiring integration

Reply:
done